packages/flame/lib/src/components/position_component.dart

```diff
@@ -192,4 +345,46 @@ abstract class PositionComponent extends BaseComponent {
       canvas.translate(-width / 2, -height / 2);
     }
   }
+} // ignore: number-of-methods
+
+class _NotifyingVector2 extends Vector2 {
```

Comment:
This is pretty smart, maybe we can have this one even more general to be used in more places and just have a callback that is called whenever the `x` and `y` is changed.

Reply:
A more general version could simply take a callback function that should be invoked when any change occurs.

Reply:
Yeah, that's what I meant. :)